src/psyclone/psyir/nodes/call.py

```diff
@@ -506,14 +487,20 @@ def _location_txt(node):
             have_codeblock = False
             while cursor:
                 if isinstance(cursor, Container):
-                    psyir = cursor.find_routine_psyir(rsym.name,
-                                                      allow_private=True)
-                    if psyir:
+                    routines = []
+                    for name in cursor.resolve_routine(rsym.name):
+                        # Since we're looking in the local Container, the
+                        # target is permitted to be private.
+                        psyir = cursor.find_routine_psyir(rsym.name,
+                                                          allow_private=True)
+                        if psyir:
+                            routines.append(psyir)
```

Comment:
I don understand why we resolve_routine but then not use the returned name. Should it be "name" what we pass to "find_routine_psyir"?

Reply:
Ah! That seems to be an error. I need to make sure this is exercised by a test.

Reply:
I've monkeypatched my way to a test for this.